src/main/haskell/kore/app/exec/Main.hs

```diff
@@ -0,0 +1,337 @@
+        <> long "pattern"
+        <> help "Kore pattern source file to verify and execute. Needs --module."
+        <> value "" )
+    <*> strOption
+        (  metavar "MODULE"
+        <> long "module"
+        <> help "The name of the main module in the Kore definition."
+        <> value "" )
+    <*> enableDisableFlag "is-program"
+        True False False
+        "Whether the pattern represents a program."
+
+
+-- | modifiers for the Command line parser description
+parserInfoModifiers :: InfoMod options
+parserInfoModifiers =
+    fullDesc
+    <> progDesc "Uses Kore definition in DEFINITION_FILE to execute pattern \
+                \in PATTERN_FILE."
+    <> header "kore-exec - an interpreter for Kore definitions"
+
+
+-- TODO(virgil): Maybe add a regression test for main.
+-- | Loads a kore definition file and uses it to execute kore programs
+main :: IO ()
+main = do
+  options <- mainGlobal commandLineParser parserInfoModifiers
+  case localOptions options of
+    Nothing -> return () -- global options parsed, but local failed; exit gracefully
+    Just KoreExecOptions
+        { definitionFileName
+        , patternFileName
+        , mainModuleName
+        , isKProgram
+        }
+      -> do
+        parsedDefinition <- mainDefinitionParse definitionFileName
+        indexedModules <- mainVerify True parsedDefinition
+        when (patternFileName /= "") $ do
+            parsedPattern <- mainPatternParse patternFileName
+            indexedModule <-
+                mainModule (ModuleName mainModuleName) indexedModules
+            mainPatternVerify indexedModule parsedPattern
+            let
+                functionRegistry =
+                    extractEvaluators Object indexedModule
+                axiomPatterns =
+                    koreIndexedModuleToAxiomPatterns Object indexedModule
+                metadataTools = constructorFunctions (extractMetadataTools indexedModule)
+                purePattern = makePurePattern parsedPattern
+                runningPattern =
+                    if isKProgram
+                        then give (sortTools metadataTools)
+                            $ makeKInitConfig purePattern
+                        else purePattern
+                expandedPattern = makeExpandedPattern runningPattern
+                finalExpandedPattern =
+                    fst $ fst $ (`runIntCounter` 1)
+                    $ do
+                        simplifiedPatterns <-
+                            ExpandedPattern.simplify
+                                metadataTools
+                                functionRegistry
+                                expandedPattern
+                        let
+                            initialPattern =
+                                case
+                                    OrOfExpandedPattern.extractPatterns
+                                        (fst simplifiedPatterns) of
+                                    [] -> expandedPattern
```

Comment:
This would mean that the pattern was simplified to bottom (i.e. an `or` with no terms is false by definition). Why would you want to continue execution?